test/distributed/org/apache/cassandra/distributed/impl/Instance.java

```diff
@@ -519,72 +524,6 @@ private Config loadConfig(IInstanceConfig overrides)
-        String partitionerName = config.getString("partitioner");
-        List<String> initialTokens = new ArrayList<>();
-        List<InetSocketAddress> hosts = new ArrayList<>();
-        List<UUID> hostIds = new ArrayList<>();
-        List<String> versions = new ArrayList<>();
-        for (int i = 1 ; i <= cluster.size() ; ++i)
-        {
-            IInstance instance = cluster.get(i);
-            IInstanceConfig config = instance.config();
-            initialTokens.add(config.getString("initial_token"));
-            hosts.add(config.broadcastAddress());
-            hostIds.add(config.hostId());
-            versions.add(instance.getReleaseVersionString());
-        }
-
-        try
-        {
-            IPartitioner partitioner = FBUtilities.newPartitioner(partitionerName);
-            StorageService storageService = StorageService.instance;
-            List<Token> tokens = new ArrayList<>();
-            for (String token : initialTokens)
-                tokens.add(partitioner.getTokenFactory().fromString(token));
-
-            for (int i = 0; i < tokens.size(); i++)
-            {
-                InetSocketAddress ep = hosts.get(i);
-                InetAddressAndPort addressAndPort = toCassandraInetAddressAndPort(ep);
-                UUID hostId = hostIds.get(i);
-                Token token = tokens.get(i);
-                String releaseVersion = versions.get(i);
-                Gossiper.runInGossipStageBlocking(() -> {
-                    Gossiper.instance.initializeNodeUnsafe(addressAndPort, hostId, 1);
-                    Gossiper.instance.injectApplicationState(addressAndPort,
-                                                             ApplicationState.TOKENS,
-                                                             new VersionedValue.VersionedValueFactory(partitioner).tokens(Collections.singleton(token)));
-                    Gossiper.instance.injectApplicationState(addressAndPort,
-                                                             ApplicationState.RELEASE_VERSION,
-                                                             new VersionedValue.VersionedValueFactory(partitioner).releaseVersion(releaseVersion));
-                    storageService.onChange(addressAndPort,
-                                            ApplicationState.STATUS_WITH_PORT,
-                                            new VersionedValue.VersionedValueFactory(partitioner).normal(Collections.singleton(token)));
-                    storageService.onChange(addressAndPort,
-                                            ApplicationState.STATUS,
-                                            new VersionedValue.VersionedValueFactory(partitioner).normal(Collections.singleton(token)));
-                    Gossiper.instance.realMarkAlive(addressAndPort, Gossiper.instance.getEndpointStateForEndpoint(addressAndPort));
-                });
-
-                int messagingVersion = cluster.get(ep).isShutdown()
```

Comment:
since we don't include `NET_VERSION` I feel that this logic should still exist here, that or update gossip to include `NET_VERSION`.

Reply:
I've added net_version. Thanks!